[18.0][MIG] automation_oca

etobella opened the pull request:
It includes new tests with Hoot for the JS

etobella committed:
[ADD] automation_oca

luisDIXMIT committed:
[IMP] automation_oca: README.md

Commit:
[UPD] Update automation_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (218 of 218 strings)

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (218 of 218 strings)

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/it/

Commit:
Added translation using Weblate (Spanish)

Commit:
Translated using Weblate (Spanish)

Currently translated at 96.7% (211 of 218 strings)

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/es/

gregory-moka committed:
Added translation using Weblate (French)

gregory-moka committed:
Translated using Weblate (French)

Currently translated at 61.9% (135 of 218 strings)

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/fr/

etobella committed:
[FIX] automation_oca: Improve selection string on ranges

gaelTorrecillas committed:
[IMP] automation_oca: Allow some expressions in domain

This commit adds the corresponding imports and preparation
code for using some variables and expressions like `today()`,
`datetime`, `user`, etc in the domain (filter) of the automation
configuration.

gaelTorrecillas committed:
[16.0] automation_oca : add demo and help for new domain options

Commit:
[UPD] Update automation_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (220 of 220 strings)

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/it/

gaelTorrecillas committed:
[IMP] automation_oca: broadcast the domain in steps

[FIX] automation_oca: pass eval_context in step domain safe_eval

OCA-git-bot committed:
[BOT] post-merge updates

gaelTorrecillas committed:
[FIX] automation_oca: pass eval_context in record step domain safe_eval

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update automation_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (220 of 220 strings)

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/it/

arielbarreiros96 committed:
[FIX] automation_oca error on record delete

Commit:
[UPD] Update automation_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (224 of 224 strings)

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/it/

etobella committed:
[IMP] automation_oca: Allow to enforce a complete different domain on an step

etobella committed:
[IMP] automation_oca: Allow to force execution with previous step finalization.

etobella committed:
[IMP] automation_oca: Allow to force scheduled date with a date of the record

Commit:
[UPD] Update automation_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/

etobella committed:
[IMP] automation_oca: Enforce verification domain on activities

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (225 of 225 strings)

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/it/

Commit:
[UPD] Update automation_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/

etobella committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (229 of 229 strings)

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (229 of 229 strings)

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/it/

etobella committed:
[IMP] automation_oca: Allow to detect when an activity is cancelled

Commit:
[UPD] Update automation_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (232 of 232 strings)

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/it/

etobella committed:
[IMP] automation_oca: Allow to delete steps

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (232 of 232 strings)

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/it/

pedrobaeza commented:
/ocabot migration automation_oca

etobella committed:
[IMP] automation_oca: fix search

OCA-git-bot committed:
[BOT] post-merge updates

etobella committed:
[IMP] automation_oca: Allow to set a context

Commit:
[UPD] Update automation_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (234 of 234 strings)

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/it/

etobella committed:
[FIX] automation_oca: Set a bigger limit for complex processes

etobella committed:
[IMP] automation_oca: allow to retry tasks

OCA-git-bot committed:
[BOT] post-merge updates

etobella committed:
[IMP] automation_oca: Set permissions properly for activities

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

macagua committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (234 of 234 strings)

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/es/

etobella committed:
[IMP] automation_oca: Use comments to apply properly the languages

luisDIXMIT reviewed (approved):
I've done some small tests locally and LGTM!!

pedrobaeza commented:
@etobella I see a bit of mix of commits from previous branches? Can you please reorder them before migration and squash what can be squashed?

etobella committed:
[FIX] automation_oca: Check do_not_wait on activation too

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (234 of 234 strings)

Translation: automation-16.0/automation-16.0-automation_oca
Translate-URL: https://translation.odoo-community.org/projects/automation-16-0/automation-16-0-automation_oca/it/

etobella committed:
[IMP] automation_oca: Allow to export

etobella commented:
@pedrobaeza This PRs comes from opened PRs that where not merged when we opened the PR. This is why we have it at the end.

pedrobaeza commented:
Yeah, but the migration commit should be the last one in this PR.

etobella commented:
I understand your needs, but on some of them I needed to make changes in order to make it work. The rebase fails if I try to make it as you suggest :sob:

pedrobaeza commented:
Yeah, but then, the commit adding the feature mixes both the feature itself and the adaptation to 18, so we will fail to distinguish the "changes to make it in 18". Do you understand me?

etobella commented:
No, while I added functionalities in other branches, I added the new functionalities on this PR. The idea was to port the changes.  It is the same that happens in a lot of other PRs: https://github.com/OCA/operating-unit/pull/600 

If I do what you want, I need to override all the history of changes of the PR.

pedrobaeza commented:
I don't get the problem you state, but well, I abstain from reviewing this one then.

etobella commented:
@pedrobaeza Let's be clear. The migration was done 3 months ago and some changes happened meanwhile where some PR where added and so on :cry: 

In any case, in order to avoid community struggles, I will do the effort to change the commit history here and make it work. However, as I am the only maintainer for now, I will try to simplify things in the future and merge with just one approval and change the way of handling it once we have more maintainers. Is it ok for you?

etobella committed:
[IMP] automation_oca: Black, isort, prettier

etobella committed:
[MIG] automation_oca: Migration to 18.0

Co-authored-by: Luís Rodríguez <luis.rodriguez@dixmit.com>

pedrobaeza reviewed (approved):
Thanks for the efforts. About the review flow from now on, as you prefer.

/ocabot merge nobump

pedrobaeza commented:
Oh, wait, the CI is red.

etobella commented:
/ocabot merge nobump

Thanks everyone for the effort.
